src/Simulation/Simulators/SparseSimulator/SparseSimulatorCS/Probes.cs

```diff
@@ -0,0 +1,360 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+// Functions that access the internal wavefunction for diagnostic purposes
+
+using Microsoft.Quantum.Simulation;
+using Microsoft.Quantum.Simulation.Common;
+using Microsoft.Quantum.Simulation.Core;
+using Microsoft.Quantum.Simulation.QuantumProcessor;
+using System;
+using System.IO;
+using System.Linq;
+using System.Runtime.InteropServices;
+using System.Text;
+
+namespace Microsoft.Quantum.SparseSimulation
+{
+
+    partial class GetAmplitudeFromInt
+    {
+        public class Native : GetAmplitudeFromInt
+        {
+            private SparseSimulator sim = null;
+            public Native(IOperationFactory m) : base(m)
+            {
+                sim = m as SparseSimulator;
+            }
+
+            public override Func<(IQArray<Qubit>, long), Microsoft.Quantum.Math.Complex> __Body__ => (args) => {
+                return sim.GetAmplitude(args.Item1, args.Item2);
+            };
+
+        }
+    }
+
+    partial class GetAmplitude
+    {
+        public class Native : GetAmplitude
+        {
+            private SparseSimulator sim = null;
+            public Native(IOperationFactory m) : base(m)
+            {
+                sim = m as SparseSimulator;
+            }
+
+            public override Func<(IQArray<Qubit>, System.Numerics.BigInteger), Microsoft.Quantum.Math.Complex> __Body__ =>  args => {
+                return sim.GetAmplitude(args.Item1, args.Item2);
+            };
+
+        }
+    }
+    public partial class SparseSimulator : QuantumProcessorDispatcher, IDisposable
+    {
+        // // Converts a big integer into a string label
+        // where the label is read over the qubits passed as an argument, in order
+        // The label for all other qubits is assumed to be 0
+        // That is: Suppose we have 11 = 1011 on qubits 0, 6,9,3.
+        // That will output the string 0001001001
+        public static string InterleaveLabelByArray(IQArray<Qubit> qubits, System.Numerics.BigInteger label)
+        {
+            // Converts a big integer into a string label
+            // where the label is read over the qubits passed as an argument, in order
+            // The label for all other qubits is assumed to be 0
+            char[] newlabel = "".PadRight((int)qubits.GetIds().Max() + 1, '0').ToCharArray();
+            string test = new string(newlabel);
+            for (int idx = 0; idx < (int)qubits.Length; idx++)
+            {
+                if (((label >> idx) & 1) == 1)
+                {
+                    int id = qubits[idx].Id;
+                    newlabel[id] = '1';
+                }
+            }
+            // std::bitset parses strings backwards
+            Array.Reverse(newlabel);
+            return new string(newlabel);
+        }
+
+        //  Returns the amplitude of a single state
+        public Microsoft.Quantum.Math.Complex GetAmplitude(string label)
```

Comment:
This and the other methods that access internal state should likely not be made public.